File: frontend/src/types/incident.ts
export interface Incident {
  id: string;
  user_id: string | null;

  platform: string;

  guild_id: string | null;
  channel_id: string | null;
  channel_name: string | null;

  prediction: string;
  probability: number;
  confidence: number;
  risk_score: number;

  message_count: number;
  conversation_excerpt: string | null;

  model_version: string;

  reviewed: boolean;
  reviewed_by: string | null;
  notes: string | null;

  created_at: string;
}

File: backend/app/schemas/incident.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class IncidentBase(BaseModel):
    """Common incident fields."""

    platform: str
    guild_id: str | None = None
    channel_id: str | None = None
    channel_name: str | None = None

    prediction: str
    probability: float
    confidence: float
    risk_score: float

    message_count: int
    conversation_excerpt: str | None = None

    model_version: str

    reviewed: bool = False
    reviewed_by: UUID | None = None
    notes: str | None = None


class IncidentCreate(IncidentBase):
    """Data required to create an incident."""

    user_id: UUID | None = None


class IncidentResponse(IncidentBase):
    """Incident returned by the API."""

    id: UUID
    user_id: UUID | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)